modules/monitoring/diagnostic_settings.tf

```diff
@@ -13,7 +13,7 @@ resource "azurerm_monitor_diagnostic_setting" "application_config_diagnostic_set
 # Key Vault Audit log monitoring
 
 resource "azurerm_monitor_diagnostic_setting" "key_vault_diagnostic_settings" {
-  count = var.key_vault_id != null ? 1 : 0
+  count = var.create_key_vault_diagnostic_settings ? 1 : 0
```

Comment:
You could update the change log with this fix

Reply:
Updated